Spigot-Server-Patches/0182-Update-VehicleCreateEvent.patch

```diff
@@ -0,0 +1,124 @@
+             EntityBoat entityboat = new EntityBoat(world, event.getVelocity().getX(), event.getVelocity().getY(), event.getVelocity().getZ());
+             // CraftBukkit end
+ 
++            // Paper start
++            if (org.bukkit.craftbukkit.event.CraftEventFactory.callVehicleCreateEvent(entityboat, block, org.bukkit.event.vehicle.VehicleCreateEvent.VehicleCreateReason.DISPENSER).isCancelled()) {
++                itemstack.count++;
++                return itemstack;
++            }
++            // Paper end
++
+             entityboat.setType(this.c);
+             entityboat.yaw = enumdirection.opposite().l();
+             world.addEntity(entityboat);
+diff --git a/src/main/java/net/minecraft/server/EntityBoat.java b/src/main/java/net/minecraft/server/EntityBoat.java
+index 01a8b84..88e9323 100644
+--- a/src/main/java/net/minecraft/server/EntityBoat.java
++++ b/src/main/java/net/minecraft/server/EntityBoat.java
+@@ -65,7 +65,7 @@ public class EntityBoat extends Entity {
+         this.lastX = d0;
+         this.lastY = d1;
+         this.lastZ = d2;
+-        this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit
++        //this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit // Paper - Relocated elsewhere
+     }
+ 
+     protected boolean playStepSound() {
+diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+index 9df7dfc..a4a2c19 100644
+--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
++++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+@@ -111,7 +111,7 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
+         this.lastY = d1;
+         this.lastZ = d2;
+ 
+-        this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit
++        //this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit // Paper - Relocated elsewhere
+     }
+ 
+     public double ay() {
+diff --git a/src/main/java/net/minecraft/server/ItemBoat.java b/src/main/java/net/minecraft/server/ItemBoat.java
+index c68e328..86f080e 100644
+--- a/src/main/java/net/minecraft/server/ItemBoat.java
++++ b/src/main/java/net/minecraft/server/ItemBoat.java
+@@ -66,6 +66,15 @@ public class ItemBoat extends Item {
+                 boolean flag1 = block == Blocks.WATER || block == Blocks.FLOWING_WATER;
+                 EntityBoat entityboat = new EntityBoat(world, movingobjectposition.pos.x, flag1 ? movingobjectposition.pos.y - 0.12D : movingobjectposition.pos.y, movingobjectposition.pos.z);
+ 
++                // Paper start
```

Comment:
Maybe move this down just before the world.addEntity(), so all properties could be used by event handlers.